cmd/vic-machine/validate/validator.go

```diff
@@ -45,314 +51,577 @@ type Validator struct {
+		// we return err directly here so we can check the type
+		return nil, err
 	}
+	if len(stores) > 1 {
+		// TODO: error about required disabmiguation and list entries in nets
+		return nil, errors.New("ambiguous datastore " + dsURL.Host)
+	}
+
+	// FIXME: commented out until components can consume moid
+	// dsURL.Host = stores[0].Reference().Value
 
-	if v.ImageStorePath != "" {
-		v.Session.Datastore, err = finder.DatastoreOrDefault(ctx, v.ImageStorePath)
-		if err != nil {
-			errs = append(errs, fmt.Sprintf("Failure finding ds (%s): %s", v.ImageStorePath, err.Error()))
-		} else {
-			log.Debugf("Found ds: %s", v.ImageStorePath)
+	return dsURL, nil
+}
+
+func (v *Validator) resourcePoolHelper(ctx context.Context, path string) (*object.ResourcePool, error) {
+	// if compute-resource is unspecified is there a default
+	if path == "" {
+		if v.Session.Pool != nil {
+			log.Debugf("using default resource pool for compute resource: %s", v.Session.Pool.InventoryPath)
+			return v.Session.Pool, nil
 		}
+
+		// if no path specified and no default available the show all
+		v.suggestAllResourcePools()
+		return nil, errors.New("no unambiguous default compute resource available so must be specified")
 	}
 
-	v.Session.Host, err = finder.HostSystemOrDefault(ctx, v.Session.HostPath)
+	// first try the path directly
+	pools, err := v.Session.Finder.ResourcePoolList(ctx, path)
 	if err != nil {
-		if _, ok := err.(*find.DefaultMultipleFoundError); !ok || !v.Session.IsVC() {
-			errs = append(errs, fmt.Sprintf("Failure finding host (%s): %s", v.Session.HostPath, err.Error()))
+		log.Debugf("failed to look up compute resource as absolute path %s: %s", path, err)
+		if _, ok := err.(*find.NotFoundError); !ok {
+			// we return err directly here so we can check the type
+			return nil, err
+		}
+
+		// if it starts with datacenter then we know we're wrong
+		if strings.HasPrefix(path, "/"+v.Session.DatacenterPath) {
+			v.suggestComputeResource(path)
+			return nil, err
 		}
-	} else {
-		log.Debugf("Found host: %s", v.Session.HostPath)
 	}
 
-	v.Session.Pool, err = finder.ResourcePoolOrDefault(ctx, v.ResourcePoolPath)
+	// assume it's a cluster specifier - that's the formal case, e.g. /cluster/resource/pool
+	// not /cluster/Resources/resource/pool
+	// everything from now on will use this assumption
+	cpath := v.computePathToInventoryPath(path)
+	log.Debugf("Converted original path %s to %s", path, cpath)
+
+	pools, err = v.Session.Finder.ResourcePoolList(ctx, cpath)
 	if err != nil {
-		errs = append(errs, fmt.Sprintf("Failure finding pool (%s): %s", v.ResourcePoolPath, err.Error()))
-		if v.Session.Datacenter == nil {
-			log.Info("Invalid datacenter. Unable to suggest valid --compute-resource")
-		} else {
-			v.suggestComputeResource()
+		log.Debugf("failed to look up compute resource as cluster path %s: %s", cpath, err)
+		if _, ok := err.(*find.NotFoundError); !ok {
+			// we return err directly here so we can check the type
+			return nil, err
```

Comment:
no, that's the error return if the finder returned an error that _wasn't_ NotFoundError - that's a different problem and indicates that there's an issue with our vmomi connection so subsequent operations are likely to fail.